CONTRIBUTING.md

```diff
@@ -0,0 +1,117 @@
+# Contributing
+
+Thank you for taking time to contribute (or at least considering to). CMS-Android is a community project as much as
+it is Crux's. 
+
+Contributions does not necessarily mean you have to modify source code and fix a bug or add a new functionality.
+Reporting a bug using the [Issues](https://github.com/crux-bphc/CMS-Android/issues) tab is also highly appreciated 
+and a good way to start contributing to the project.
+
+It is essential that you know how to use Git as well as GitHub. While the maintainers will be more than happy helping
+you make your first PR, prerequisite knowhow makes it easier for everyone involved. Check this 
+[guide](https://github.com/firstcontributions/first-contributions)  if you've never contributed to a project on 
+GitHub before.
+
+Please note we have a [Code of Conduct](code-of-conduct.md). Please follow it in all your interactions with the project.
+
+When you submit code changes, your submissions are understood to be under the same [MIT License](LICENSE) that covers the 
+project. Feel free to contact the maintainers if that's a concern.
+
+## Moodle Endpoints Used By The App
+
+The Moodle endpoints used by this app have been collected in a [Postman](https://getpostman.com") workspace 
+[Here](https://www.getpostman.com/collections/e2c0439f144f7d3f60ed). You may import this collection into Postman directly using the link. 
+
+## Contributors Pull Request Checklist
+
+1. Fork the repository and branch from `development`.
+
+2. Work on your changes. Make sure you commit as frequently as required. Once you're done making the changes, push the commits
+   to your fork. Checkout the **Commit Messages** section to find guidelines relating to commit messages.
+
+3. If the changes you make fixes a bug in `master` or adds a new feature, note it down in the [changelog](CHANGELOD.md).
+   
+3. Head over to the [pull requests](https://github.com/crux-bphc/CMS-Android/pulls) page and create a new pull request. Make sure that the PR
+   is from your feature branch into this repo's `development` branch. If you are making changes that affect the UI in anyway, attach an image
+   in your PR description.
+
+4. Wait for a mainter to review your commit. If they finds any issue (including redundant code, unnecessary changes, potential bugs),
+   they will point it out to you. Make any necessary changes and push them to your fork. Ideally, try not to force push as it becomes
```

Comment:
This can segue into its own conversation about the workflow. Do you want people to create `FIXUP: further changes` commits that you fixup before finally rebasing/merging? That is one of the ways to keep the commit history in order while allowing people to forcepush at will.

Reply:
Interesting. I've never seen FIXUP commits tbh. This is something was troubling me for the previous couple of PRs made to the repo. Granularity is lost if PRs are squashed so that fixup (though not tagged as such) commits do make it into the repo. Will GitHub automatically squash commits annotated with !fixup (or alternatively FIXUP) in the commit subject? Or does the commits have to be squashed using `--auto-squash` and then force pushed? Either way, this might be something important to note down in the guidelines.

Reply:
Let's go meta and discuss this current PR. You have added some commits like "Fix typos" or "Hard line wrap" or the biggest offender "Fix problems caught in review".

In that last one, you fixed one missed `s/CRUx/Crux/` change, which should have been a part of the commit where you proclaim revolution (nice, BTW :P). 

Either you should have gone and edited the first commit itself or have made that second change in a commit: `FIXUP: The crux commit`. Now, after review, its simple to locally to `git rebase -i master` and fixup all the commits you've marked as such, before pushing and finally rebasing the branch to master, thus closing the PR.

Reply:
Ahh makes sense. I was hoping GitHub would automatically apply all fixups while rebasing. I'll add a new section to the guidelines to detail how this should be done.

Reply:
I've added a section called **Multi Commit PRs** that detailing how to create fixup commit using Git's `!fixup` annotation `git rebase --autosquash` option.